app/controllers/lifehacks_controller.rb

```diff
@@ -7,4 +7,32 @@ def show
     @lifehack = Lifehack.find(params[:id])
     @reviews = @lifehack.reviews
   end
+
+  def new
+    if user_signed_in?
+      @lifehack = Lifehack.new
+    else
+      flash[:error] = "You must be signed in to add a lifehack!"
+      redirect_to lifehacks_path
```

Comment:
Nice job with the logic here. With devise, you can use a before action at the top of the controller that accounts for this instead. It will also display a standard flash message letting the user know they must log in first. The before action would go on line two of this controller and can look something like this: 
```
  before_action :authenticate_user!, except: [:index]
```